recipe/meta.yaml

```diff
@@ -1,32 +1,38 @@
+{% set name = "bzip2" %}
 {% set version = "1.0.8" %}
 
 package:
-  name: bzip2
+  name: {{ name }}
   version: {{ version }}
 
 source:
-  url: https://sourceware.org/pub/bzip2/bzip2-{{ version }}.tar.gz
+  url: https://sourceware.org/pub/{{ name }}/{{ name }}-{{ version }}.tar.gz
   sha256: ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269
   patches:
-    - 0001-cross.patch
+    - 0001-cross.patch  # [not win]
+    # 2024/2/23: build_dll.patch might not be required in future versions.
+    # Check it before building the next release.
+    - build_dll.patch   # [win]
 
 build:
-  number: 0
+  number: 5
   run_exports:
     - {{ pin_subpackage('bzip2') }}
 
 requirements:
   build:
     - {{ compiler('c') }}
-    - m2-patch  # [win]
-    - m2-gcc-libs  # [win]
-    - patch  # [not win]
-    - jom   # [win]
-    - make  # [unix]
+    - m2-patch             # [win]
+    - patch                # [not win]
+    - make                 # [unix]
 
 test:
-  commands:
-    - bzip2 --help  # [unix]
+  files:                   # [win]
+    - bz2.c                # [win]
+  requires:                # [win]
+    - {{ compiler('c') }}  # [win]
+  commands:                                 # [unix]
+    - bzip2 --help                          # [unix]
```

Comment:
```suggestion
    - bzip2 --help
```

Reply:
Funny but if your `meta.yaml` has this wrongly placed selector `# [unix]`:
```
test:
  commands:                                 # [unix]
    - bzip2 --help
```
Then  you will get an error in the host environment on `win-64`:
`InvalidVersionSpec: Invalid version '--help': empty version component`